### PR TITLE
Fix clippy warnings

### PR DESCRIPTION
**Pull request by mikevoronov**

### CODE REVIEW

File: tools/cli/air-trace/src/stats.rs [hunk 1]
```diff
@@ -48,6 +48,7 @@ pub(crate) fn stats(mut args: Args) -> anyhow::Result<()> {
 
     let mut stats = self::report::StatsReport::new();
 
+    #[allow(clippy::significant_drop_in_scrutinee)]
```
Comment (monoid) [inline tools/cli/air-trace/src/stats.rs:51]: One can avoid this warning by passing `&mut stdin` instead (making `mut stdin` above).
I don't see it in my installation, BTW.

File: crates/air-lib/air-parser/src/ast/instruction_arguments.rs [hunk 2]
```diff
@@ -25,7 +25,7 @@ use air_lambda_ast::LambdaAST;
 use serde::Deserialize;
 use serde::Serialize;
 
-#[derive(Debug, PartialEq, Clone, Serialize, Deserialize)]
+#[derive(Debug, PartialEq, Eq, Clone, Serialize, Deserialize)]
```
Comment (monoid) [inline crates/air-lib/air-parser/src/ast/instruction_arguments.rs:28]: Is adding `Eq` solves some warning, or "just in case"?

**Comment by monoid** [comment]
utACK

**Review by monoid — APPROVED** [review]

File: crates/air-lib/air-parser/src/ast/instruction_arguments.rs (hunk 2 above)
Comment (mikevoronov) [reply crates/air-lib/air-parser/src/ast/instruction_arguments.rs:28]: Yep, it solves `clippy` warning.

File: tools/cli/air-trace/src/stats.rs (hunk 1 above)
Comment (mikevoronov) [reply tools/cli/air-trace/src/stats.rs:51]: Looks like the reason here is `anyhow::Result`.

Comment (monoid) [reply tools/cli/air-trace/src/stats.rs:51]: The lint seems to be broken:
https://github.com/rust-lang/rust-clippy/issues/8987

**Pull request merged**